src/Commands/BuildToolsBase.php

```diff
@@ -1003,12 +1003,15 @@ protected function getLatestWorkflow($site)
     public function getBuildMetadata($repositoryDir)
     {
         return [
-          'url'         => exec("git -C $repositoryDir config --get remote.origin.url"),
-          'ref'         => exec("git -C $repositoryDir rev-parse --abbrev-ref HEAD"),
-          'sha'         => $this->getHeadCommit($repositoryDir),
-          'comment'     => exec("git -C $repositoryDir log --pretty=format:%s -1"),
-          'commit-date' => exec("git -C $repositoryDir show -s --format=%ci HEAD"),
-          'build-date'  => date("Y-m-d H:i:s O"),
+          'url'           => exec("git -C $repositoryDir config --get remote.origin.url"),
+          'ref'           => exec("git -C $repositoryDir rev-parse --abbrev-ref HEAD"),
+          'sha'           => $this->getHeadCommit($repositoryDir),
+          'comment'       => exec("git -C $repositoryDir log --pretty=format:%s -1"),
+          'commit-date'   => exec("git -C $repositoryDir show -s --format=%ci HEAD"),
+          'build-date'    => date("Y-m-d H:i:s O"),
+          'git-provider'  => $this->git_provider->getServiceName(),
+          'ci-provider'   => $this->ci_provider->getServiceName(),
+          'api-host-host' => $this->git_provider->api()->getApiHost(),
```

Comment:
Presume this should just be `api-host` and not `api-host-host`?

I'm unsure whether we should be adding this here. It is a well-known value. It seems unlikely that a script could use it without knowing which service it is for (agnostic of service, that is), as the API calls are going to vary by service. Rather than provide this constant, it might be better to provide a small library sites could link to if they needed to make an API call (probably as a future task).

Reply:
I think the primary purpose of putting it here was to handle those instances where people had non-standard URLs (for example, self-hosted GitLab). While we could extract it out of the Git URL for that particular service, it seemed like we would be duplicating code & effort when that time came. It didn't seem to introduce a ton of extra overhead to add it here and it means it can be used in the future without worry.

Reply:
Ah, for GitLab custom API servers that makes sense.